app/src/main/java/com/animal_villa/dto/Settings.kt

```diff
@@ -0,0 +1,14 @@
+package com.animal_villa.dto
+
+import androidx.room.Entity
+import com.animal_villa.dao.ISettingsDAO
+
+@Entity
+class Settings {
+    var volume = 1
+    var textSpeed = 1
+    var soundEffects = true
+    fun updateSettings(dao: ISettingsDAO) {
+        dao.updateSettings(this)
+    }
```

Comment:
A data class/DTO/noun class should not invoke a DAO to save itself. Data class/noun class/DTO should only have attributes.  The business logic, persistence logic, etc., should be handled by a different class.